src/cmdline/cmdline_input.ts

```diff
@@ -0,0 +1,237 @@
+import { QuickPick, QuickPickItem, ThemeIcon, window, Disposable, QuickInputButton } from "vscode";
+
+import { disposeAll } from "../utils";
+import { calculateInputAfterTextChange } from "../utils/cmdline_text";
+import { createLogger } from "../logger";
+
+const logger = createLogger("CmdLineInput", false);
+
+export interface CmdlineInputListeners {
+    onAccept: () => void;
```

Comment:
You don't need this callback bolierplate. 


[`19e1412` (#2001)](https://github.com/vscode-neovim/vscode-neovim/pull/2001/commits/19e1412c4942f2b0eac9136f77329a5fc6e44e5a)

Reply:
You're right, you don't need it. I did this so we could decouple the interactions with neovim from the input box itself